src/tests/test_main.cpp

```diff
@@ -22,9 +23,31 @@ int main(int argc, const char** argv) {
 
     core::CrashHandler crash_handler;
 
-    CommandLineArguments args(argc, argv);
+    /* Check wether "-vv" option is set.
+     * If yes then change it to "-v" and remember this in "moreVerbose" variable.
+     */
 
-    if (args.parse(NULL) && args.isVerbose()) {
+    /*
+     * Reason of changing "-vv" to "-v" is that we also want to instruct
+     * CppUTest to give more verbose output
+     */
+
+    bool moreVerbose = false;
+    const char** argv2 = new const char*[(unsigned int)argc];
+    for (int i = 0; i < argc; i++) {
+        if (strcmp(argv[i], "-vv") == 0) {
+            argv2[i] = "-v";
```

Comment:
I don't think this is a good approach. `-vv` is a special case of the generic feature of mixing flags; if we support `-vv`, I'd expect that we also support any mixed combination, so that these calls would be equivalent:

```
./roc-test-core -vvc
./roc-test-core -cvv
./roc-test-core -vcv
./roc-test-core -vc -v
./roc-test-core -cv -v
...
```

(and this example mentions only `-c` option, while there are other options too)

This would require a much more complicated option parsing than just replacing one string with another.

On the other hand, CppUTest itself doesn't support this feature, so e.g. `-vc` and `-cv` aren't accepted, and only `-c -v` or `-v -c` are accepted.

So instead of providing incomplete (and thus surprising for the user) support of mixing flags, maybe just provide a different flag without mixing support?

E.g. we can add a new flag `-t` for our feature and expect that it will be always provided as a separate entry in argv.

Reply:
Ok. I will implement seperate flag `-t`. 

Is there any other suggestion regrading implementation?

Reply:
Nope. Another bad thing with current implementation is using bare new/delete instead of RAII (e.g. core::Array), but the new implementation will be different, I guess, so I didn't mention it.

Reply:
Let me discuss my new approach briefly.
I will assume that `-t` flag will be provided seperately for enabling trace. Now because `CPPUTEST` does not recognise that option but we still want to instruct it to enable verbose log we have to pass `-v` flag to it. So I will just change `-t` to `-v`. 

But this might create a problem of duplication of option when there is already a `-v` flag provided by user.

Reply:
Let me discuss my new approach briefly.
I will assume that `-t` flag will be provided seperately for enabling trace. Now because `CPPUTEST` does not recognise that option but we still want to instruct it to enable verbose log we have to pass `-v` flag to it. So I will just change `-t` to `-v`. 

But this might create a problem of duplication of option when there is already a `-v` flag provided by user. I tried to provide duplicate flags to CPPUTEST in our original version of codes and indeed it works fine as of now. But in future they might change the behaviour to handle this scenario. So please suggest whether I should implement it this way or make sure that `CPPUTEST` will never get duplicate option because of this feature.

Reply:
I think we can leave duplicate `-v` flags, this is an expected behavior and it's unlikely that they'll break it. So just replacing `-t` with `-v` should be fine.